wsmaster/che-core-api-workspace-activity/src/main/java/org/eclipse/che/api/workspace/activity/WorkspaceActivity.java

```diff
@@ -26,8 +26,14 @@
 @Table(name = "che_workspace_activity")
 @NamedQueries({
   @NamedQuery(
-      name = "WorkspaceActivity.getExpired",
+      name = "WorkspaceActivity.getExpiredIdle",
       query = "SELECT a FROM WorkspaceActivity a WHERE a.expiration < :expiration"),
+  @NamedQuery(
+      name = "WorkspaceActivity.getExpiredRunTimeout",
+      query =
+          "SELECT a FROM WorkspaceActivity a WHERE "
+              + "(a.status = org.eclipse.che.api.core.model.workspace.WorkspaceStatus.RUNNING AND "
+              + "a.lastRunning < :timestamp - :runTimeout)"),
```

Comment:
- select  a.workspaceId? I think we can make same replacement in  WorkspaceActivity.getExpiredIdle too"

-  replace two parameters query  `:timestamp - :runTimeout` with one parameter `:X` do not ask the database to make additional calculations that are not needed.
Can you provide `explain` of this query, please? Do we need an index on `a.status, a.lastRunning`?

Reply:
I will change to return a list of strings instead of `WorkspaceActivity`, and will pass in the difference to the query.

See the above explain plan.  The existing indexes should be fine.